### PR TITLE
build: use docker buildx to build images

### PR DESCRIPTION
**Pull request by sarandia**
Updated the build-release file to use docker buildx. This way, we
can natively cross-compile multi-arch docker images. Updated
dockerfiles/Dockerfile.arm64v8 to use native cross-compilation instead
of QEMU emulation.

Docker buildx is compatible with docker build commands, so existing
builds should not be affected.

#3119

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [NA] Example configuration file for the change
- [NA] Debug log output from testing the change

- [NA ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

This change is about the configuration file(s) so no example configuration is needed.

**Documentation**

- [NA] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by sarandia** [comment]
@niedbalski Here's the PR following our discussion in #3152. PTAL. Thank you!

**Comment by sarandia** [comment]
Rebased on master

**Comment by sarandia** [comment]
@niedbalski Any idea why the docker build integration test times out? It runs docker build on the x86 Dockerfile, which shouldn't be related to this change.

**Comment by niedbalski** [comment]
Running the integration tests again, @sarandia we'll see how that goes.

File: .github/workflows/build-release.yaml [hunk 1]
```diff
@@ -75,7 +75,18 @@ jobs:
       max-parallel: 48
       fail-fast: true
       matrix:
-        arch: [ x86_64, arm64v8, arm32v7, x86_64-debug ]
+        arch: [ amd64, arm64v8, arm32v7 ]
+        suffix: [ x86_64, arm64v8, arm32v7, x86_64-debug ]
+        hostos: [ linux ]
```
Comment (niedbalski) [inline .github/workflows/build-release.yaml:80]: rename it to os

Comment (sarandia) [reply .github/workflows/build-release.yaml]: Done.

**Comment by niedbalski** [comment]
Holding this until PR #3225 gets merged.

**Pull request merged**